### PR TITLE
Regular Polygon feature

### PR DESCRIPTION
**Pull request by shanranm**
Addresses #31

### CODE REVIEW

**Comment by sthewissen** [comment]
I'm not a fan of the `IsRegular` property. I think a cleaner solution is making `Sides` a default value of 4. Wherever `IsRegular` is used would then become a check for `Sides` being 4 or not. 

Also perhaps an additional check needs to be added that throws an `ArgumentException` when sides is not 3 or more. There's already a similar `ArgumentException` for background gradient angle which needs to be 0-360. Does this also apply for your OffsetAngle?

**Comment by shanranm** [comment]
As I've mentioned on the issue, sides by default is `4`, but `IsRegular` switches it from begin a Rectangle to regular polygon - Square
Will add the checks on `Sides`. OffsetAngle, well its an angle, can be anything, above 360 means -  `value % 360`. Will add checks just to be in consistent with gradient angle.

**Comment by sthewissen** [comment]
One of the things I've been struggling with lately is figuring out a way to check whether or not all of the supported properties work and work when combined with one another. Adding polygons adds a whole new dimension to that as well. A lot of scenarios to support. The `DebugPage` was a start for that, but maintaining that manually becomes a chore. Will look into something coming up with something for this myself.

**Comment by shanranm** [comment]
I was also thinking on the same lines.. hence the new Polygon try out page. Still needs a lot of work in adding toggle for all the features...

**Comment by sthewissen** [comment]
Yeah, I really like that idea. Will try to expand upon that too for the normal pancakes.

**Comment by sthewissen** [comment]
Do you mind if I start expanding on your idea for the debugging page for the "normal" PancakeView? Perhaps we can at some point merge them into one big debugging page with all the options?

image

**Comment by shanranm** [comment]
Sure, please go ahead!
Love the monkey face!!

**Comment by sthewissen** [comment]
If all is well, this new debug page should now be feature complete. Some properties aren't working, which is nice to see, because that tells me where I/we still need to do some work 😂

image

**Comment by shanranm** [comment]
Wow, now THIS is a Debug Page!
I can see lots of things that are not supposed to happen 😄 
Just to be safe I reverted back to master and opened this page, some things are real messed up!! Especially with gradients. I guess no one's changing gradient at runtime.
We need to fix them before even thinking of merging this PR 😞

**Comment by sthewissen** [comment]
I've been testing on iOS and I think quite a bit of things are working better than I had actually expected 😅That'll probably be my first focus, since I know the iOS code better.

File: src/Xamarin.Forms.PancakeView.Multi/Platforms/iOS/PancakeViewRenderer.cs [hunk 1]
```diff
@@ -14,444 +14,436 @@
...
+			cornerPath.AddLineToPoint(insetBounds.X, (float)(insetBounds.Y + topLeft));
+			cornerPath.AddArc((float)(insetBounds.X + topLeft), (float)(insetBounds.Y + topLeft), (float)topLeft, (float)Math.PI, (float)(Math.PI * 1.5), false);
+
+			return cornerPath;
+		}
+
+		public void AddLayer(CALayer layer, int position, UIView viewToAddTo)
+		{
+			// If there is already a layer with the given name, remove it before inserting.
+			if (viewToAddTo.Layer.Sublayers != null && layer != null)
```
Comment (shanranm) [inline src/Xamarin.Forms.PancakeView.Multi/Platforms/iOS/PancakeViewRenderer.cs]: should go in even when sublayers are null. Means it is the first time its adding layer

Comment (sthewissen) [reply src/Xamarin.Forms.PancakeView.Multi/Platforms/iOS/PancakeViewRenderer.cs]: Fixed. That was a remnant from some layering rework I did, similar to #33.

**Comment by sthewissen** [comment]
@shanranm - I fixed almost all the iOS rendering bits. The only scenario that is still giving problems right now is the non-square + gradient border + small corner radius at some rotations. Gradient borders were a tough nut to crack and I've had to give up on `BorderDrawingStyle` for now.

image

**Comment by shanranm** [comment]
And that sir, should fix everything.
well, of course leaving out the `BorderDrawingStyle`

**Comment by sthewissen** [comment]
You mean fix everything on both platforms or just iOS? I just tested iOS and that works like a charm! Thanks for all the hard work so far, I really appreciate it. Couldn't have done it without you 😃

Moving on to Droid; Running the debug app on Android on a GenyMotion emulator I get some really weird behavior. Not sure if that's a good test case though, but I don't have my physical Android device near. All the sliders in there are also kind of behaving weirdly 😂 Just some of the things I see:

- BackgroundGradient not working.
- Dashed border behaving weirdly.
- Clipping + border produces no border.
- Sides/offset angle seem to misbehave.

image

**Comment by shanranm** [comment]
I swear I fixed on Android too 😄 
May be dependent on Android version?
I did debug on a physical device with version - 8.1.0
Still, will take a look again

**Comment by sthewissen** [comment]
Yeah, like I said, I'm not sure how good my test setup is for this. I'll have access to a physical device tomorrow, so will also recheck then.

**Comment by shanranm** [comment]
I was able to reproduce all of them on an emulator.
For some weird reason, `Draw` is not called on `PancakeDrawable` when `BackgroundColor` is changed. To handle this, new drawable is created (also disposing the older one)

**Comment by sthewissen** [comment]
Cool, that last update fixes a lot! Is it just me though or are these sliders rendering very weirdly and  very un-userfriendly (no track, very hard to control)?

image

Also, seems like the background bleeds out from under the border in some cases?

image

I also just fixed a small bug where there would be a minuscule border when it's set to 0.

**Comment by shanranm** [comment]
Yea, sliders have issue on Android, had to manually set `WidthRequest` for debugging.
Just set it as style, will be applied to all elements on the page.
The background leaking have always been there... i'd like to rewrite another create path specifically for `sides == 4` case... may be another PR?

**Comment by sthewissen** [comment]
Agreed. Let's merge this one!

Once again though, let me make clear that I really appreciate what you've done in this PR! It truly is an awesome addition 😃

**Comment by shanranm** [comment]
It was my pleasure!! 😄 
I don't think it would've been come this far without your active and continuous support!!

**Pull request merged**